starknet-replay/src/histogram/plot.rs

```diff
@@ -0,0 +1,102 @@
+//! This module uses the library `plotter` to render and export the histogram.
+//!
+//! Some settings are hardcoded like:
+//!  - Size and font of labels
+//!  - Colours
+//!  - Export format
+//!
+//! To use another plotting library, it's sufficient to rewrite the function
+//! `render`.
+
+use std::path::PathBuf;
+
+use plotters::backend::SVGBackend;
+use plotters::chart::ChartBuilder;
+use plotters::coord::ranged1d::{IntoSegmentedCoord, SegmentValue};
+use plotters::drawing::IntoDrawingArea;
+use plotters::series::Histogram;
+use plotters::style::full_palette::{RED, WHITE};
+use plotters::style::text_anchor::{HPos, Pos, VPos};
+use plotters::style::{Color, FontTransform, IntoFont, TextStyle};
+
+use crate::histogram::{Config, HistogramError};
+use crate::ReplayStatistics;
+
+/// Create and export the histogram as SVG file.
+///
+/// # Arguments
+///
+/// - `filename`: The filename of the exported SVG file.
+/// - `title`: The title of the histogram.
+/// - `config`: The configuration object of the histogram.
+/// - `libfunc_stats`: The input data to be plotted.
+///
+/// # Errors
+///
+/// Returns [`Err`] if:
+///
+/// - There is an error computing the histogram `Config` object.
+/// - There is an error rendering the histogram.
+/// - There is an IO error saving the SVG file of the histogram.
+pub fn render(
+    filename: &PathBuf,
+    title: &str,
+    config: &Config,
+    libfunc_stats: &ReplayStatistics,
+) -> Result<(), HistogramError> {
+    let list_of_libfuncs = libfunc_stats.get_libfuncs();
+    let root = SVGBackend::new(filename, (config.width, config.height)).into_drawing_area();
+
+    root.fill(&WHITE)?;
+
+    // Putting spaces in the caption creates panic
+    // https://github.com/plotters-rs/plotters/issues/573#issuecomment-2096057443
+    let mut chart = ChartBuilder::on(&root)
+        .x_label_area_size(config.x_label_area)
+        .y_label_area_size(150)
+        .margin(30)
+        .caption(title, ("sans-serif", 50.0))
```

Comment:
We don't seem to handle that potential panic though? For now I would issue an error within the code and submit an issue to track fixes for this.

Reply:
That's not valid anymore with the newer version of `plotters`